### PR TITLE
Add Ruby 3.1 and Rails 7 to the build matrix

### PR DESCRIPTION
**Pull request by rzane**
### Summary

> Provide a general description of the code changes in your pull
request... were there any bugs you had fixed? If so, mention them. If
these bugs have open GitHub issues, be sure to tag them here as well,
to keep the conversation linked together.

### Other Information

> If there's anything else that's important and relevant to your pull
request, mention that information here. This could include
benchmarks, or other information.

> Thanks for contributing to TestTrack!

/domain @Betterment/test_track_core
/platform @Betterment/test_track_core

### CODE REVIEW

File: Appraisals [hunk 1]
```diff
@@ -5,3 +5,7 @@ end
 appraise 'rails-6-1' do
   gem 'railties', '~> 6.1.0'
 end
+
+appraise 'rails-7.0' do
```
Comment (smudge) [inline Appraisals:9]:
```suggestion
appraise 'rails-7-0' do
```

Comment (rzane) [reply Appraisals:9]: Whoops! Seems to do the same thing, but I'll fix this for consistency!

Comment (smudge) [reply Appraisals:9]: It's generating slightly different filenames, yeah -- `7.0.gemfile` vs `7_0.gemfile`

**Review by smudge — APPROVED** [review]
domain LGTM && platform LGTM

**Pull request merged**